astropy/utils/data.py

```diff
@@ -894,21 +914,38 @@ def _find_pkg_data_path(data_name, package=None):
         else:
             package = module.__package__
     else:
+        # package errors if it isn't a str
+        # so there is no need for checks in the containing if/else
         module = resolve_name(package)
 
-    rootpkgname = package.partition('.')[0]
-
-    rootpkg = resolve_name(rootpkgname)
-
+    # module path within package
     module_path = os.path.dirname(module.__file__)
-    path = os.path.join(module_path, data_name)
+    full_path = os.path.join(module_path, *path)
 
+    # Check that file is inside tree.
+    rootpkgname = package.partition('.')[0]
+    rootpkg = resolve_name(rootpkgname)
     root_dir = os.path.dirname(rootpkg.__file__)
-    if not _is_inside(path, root_dir):
+    if not _is_inside(full_path, root_dir):
         raise RuntimeError(f"attempted to get a local data file outside "
                            f"of the {rootpkgname} tree.")
 
-    return path
+    return full_path
+
+
+def _find_pkg_data_path(data_name, package=None):
```

Comment:
Is this for backwards compatibility, even though it was private? If so, maybe at least deprecate it?

Reply:
Hah! Erik and I went back on forth on this one. @nstarman must have added/removed it several times by now. I asked him to remove the deprecation because it is a hidden function. If we really don't like this, it is better to just remove the function. Or we can keep it around silently and remove in 5.0. But I don't think there is much point in deprecating a hidden function.

Reply:
Ah, I see! Starting from scratch without thinking there must be a reason this is being kept, I'd just remove it - it is private and no-one should rely on it.

Reply:
sounds good!

Reply:
> no-one should rely on it.

Haha. Yeah. I have been. Hence this PR.